app/enterprise/1.3-x/developer-portal/theme-customization/single-page-app.md

```diff
@@ -0,0 +1,142 @@
+
+To view the basic example Angular template from this guide visit the [`example/spa-angular`](https://github.com/Kong/kong-portal-templates/tree/example/spa-angular) branch from the kong-portal-templates branch.
+
+### Prerequisites
+
+* Kong Enterprise 1.3 or later
+* Portal Legacy is turned off
+* The Developer Portal is enabled and running
+* kong-portal-cli tool is installed locally
+
+### What is a SPA
+
+A Single Page App (SPA) is a website that loads all HTML, Javascript, and CSS on the first load. Instead of loading subsequent pages from the server, javascript is used to dynamically change the page content. You may wish to use a SPA in Dev Portal if you have a preexisting SPA you want to integrate with portal, or you are trying to achieve a more application like experience across many pages. A SPA takes control of routing from the server, and handles it client-side instead.
+
+If you simply wish to add interactive javascript to a few pages, it may make more sense to add the javascript to only layouts that require it, and maintain server side rendering. [Learn more](/enterprise/{{page.kong_version}}/developer-portal/theme-customization/adding-javascript-assets) about adding Javascipt to a layout without implementing a SPA.
+
+### Making Choices
+
+
+We recommend Catalog and Spec routes not be handled by SPA
+If you are using Authentication, then you probably also want to leave server-side rendering for any account pages
+
+### Getting Started
+Clone down the portal-templates repo
+
+Create a file called `router.conf.yaml` in `workspaces/default` This file will override the default routing, allowing you to control routing via javascript.
+
+`router.conf.yaml` must be a yaml file, where the key is each route, and the value a content or spec path. `/*` Is a catch-all wildcard for all routes not specified in `router.conf.yaml`, it will overwrite all default routing set by collections or set in headmatter.
+
+In the `router.conf.yaml` example below, we are hardcoding routing for all kong related functionality and deferring to SPA routing for all other routes.  The `/*` route at the bottom of the file is a wildcard route. The wildcard route, in this case, tells Kong to serve `content/index.txt' for any request that is not handled by the route declarations above.
+
+```
+/login: content/login.txt
+/logout: content/logout.txt
+/register: content/register.txt
+/settings: content/settings.txt
+/reset-password: content/reset-password.txt
+/account/invalidate-verification: content/account/invalidate-verification.txt
+/account/resend-verification: content/account/resend-verification.txt
+/account/verify: content/account/resend-verification.txt
+/documentation: content/documentation/index.txt
+/documentation/httpbin: specs/httpbin.json
+/documentation/petstore: specs/petstore.yaml
+/*: content/index.txt
+
+```
+
+### Creating Your SPA
```

Comment:
Can you change this section to numbered steps with explicit commands?

Reply:
Step 1. Creating the SPA
1. 
2.
Step 2. Mounting a SPA

Step 3. Loading your Layout

Step 4. Deploy to Portal

(something like that)

Reply:
@levi I added a numbered list of the steps. Let me know if this is sufficient.